File: neural java/Node.java
package itu.ejuu.neural;

import java.util.ArrayList;

public class Node {
	
	public static double L = 1.0; // learning rate
	
	public ArrayList<Connection> conn_in, conn_out;
	private double inputValue = 0.0;
	private double outputValue = 0.0;
	private double errorValue = 0.0;
	private double bias = 1.0;

	public Node(double inititalBias){
		bias = inititalBias;
		conn_in = new ArrayList<Connection>();
		conn_out = new ArrayList<Connection>();
	}
	
	public void setInput(double input){
		this.inputValue = input;
	}
	
	public double getOutput(){
		return outputValue;
	}
	
	public double getError(){
		return errorValue;
	}
	
	public double forwardOperation(){
		if(conn_in.size() == 0){ // input node
			outputValue = inputValue;
			return inputValue;
		}
		
		double S = bias;
		for(Connection c : conn_in){
			S += c.getFromNode().forwardOperation() * c.getWeight();
		}
		
		double a = 1.0/(1.0 + Math.pow(Math.E, -1.0 * S));
		outputValue = a;
		return a;
	}
	
	public void computeError(double T){
		double O = this.outputValue;
		
		if(conn_out.size() == 0){ // output node
			errorValue = O*(1-O)*(T - O);
			return;
		}
		// layer node (warning: requires next layer to be calculated first
		double nextSum = 0.0;
		for(Connection c : conn_out){
			nextSum += c.getToNode().getError() * c.getWeight();
		}
		errorValue = O*(1-O) * nextSum;
	}
	
	public void updateBias(){
		double delta = Node.L*this.getError();
		bias = bias + delta;
	}

	public double getBias() {
		return bias;
	}
}

File: DataMining/Neural Networks/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace DataMining.Neural_Networks
{
    public class Node
    {
        public const double LearningRate = 0.5;

        public List<Connection> InCon, OutCon;

        public double InputValue { private get;  set; }
        public double OutputValue { get; private set; }
        public double ErrorValue { get; private set; }
        public double Bias { get; private set; }


        public Node(double initialBias)
        {
            InputValue = 0.0;
            OutputValue = 0.0;
            ErrorValue = 0.0;
            Bias = 1.0;
            
            Bias = initialBias;

            InCon = new List<Connection>();
            OutCon = new List<Connection>();
        }

        public double ForwardOperation()
        {
            if (InCon.Count == 0) // input node
            {
                OutputValue = InputValue;
                return InputValue;
            }

            //Console.WriteLine("Forward Op: " + this.GetHashCode());

            double S = Bias; //+ InCon.Sum(conn => conn.From.ForwardOperation()*conn.Weight);
            foreach (Connection c in InCon)
            {
                double value = c.From.ForwardOperation() * c.Weight;
                //Console.WriteLine("\t From: " + conn.From.GetHashCode() + " = "+value);
                S += value;
            }

            double A = 1.0/(1.0 + Math.Pow(Math.E, -1.0 * S));
            OutputValue = A;
            //Console.WriteLine("\t A = " + A);
            return A;
        }

        public void ComputeError(double T)
        {
            double O = OutputValue;

            if (OutCon.Count == 0) // output value
            {
                ErrorValue = O*(1.0 - O) * (T - O);
                return;
            }

            //layer mode
            double nextSum = OutCon.Sum(c => c.To.ErrorValue * c.Weight);
            ErrorValue = O * (1.0 - O) * nextSum;
        }

        public void UpdateBias()
        {
            Bias = Bias + (Node.LearningRate*ErrorValue);
        }
    }
}